### PR TITLE
Feature reserve data

### PR DESCRIPTION
**Pull request by UbdaNam**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x] Preserve data
- [x] Add save in to file methods
- [x] Add load from file methods

## Description of the tasks completed 👇 

- [x] Preserve data
- [x] Install ruby
- [x] Use the linter tool that analyzes my source code to flag programming errors, bugs, stylistic errors, and suspicious constructs. 💯 
  
## How should it be tested

Clone the repository
```
git clone https://github.com/UbdaNam/OOP-school-library.git
```
Navigate to the folder
```
cd OOP-school-library
```
Checkout the branch
```
git checkout feature_reserve_data
```
Run the program
```
ruby main.rb
```

### CODE REVIEW

File: modules/load_data.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+module LoadData
+  def load_books
+    books = JSON.parse File.read('./data/books.json')
+    return if books.nil?
+
+    books.each do |book|
```
Comment (OyePriscilla) [inline modules/load_data.rb:6]: - [ ]  I got this error message after I deleted books.json file and tried to start up the app. Perhaps you need to cater for a situation whereby the database is empty. 👇 

Untitled

## Project requirement

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @UbdaNam and @BANCUNGUYE66 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Preserved the data from app ✔️
App loaded preserved data on startup ❎

## Required Changes ♻️
 Kindly ensure that you fix the broken link in the` Author (2nd author Linkedln)` section of the README file. This will make your README file to be more engaging. 👍

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/load_data.rb (hunk 1 above)
Comment (UbdaNam) [reply modules/load_data.rb:8]: thank you @OyePriscilla for your request, will make the changes and request for rereview 👍

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @UbdaNam @BANCUNGUYE66 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
Preserved the data from app ✔️
App loaded preserved data on startup ❎

 ## Required Changes ♻️
_Check the comments under the review._
as suggested previously  Kindly ensure that you fix the broken link in the Author (2nd author Linkedln) section of the README file. This will make your README file to be more engaging. 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by UbdaNam** [comment]
hi @alexander16108 can you elaborate on your second highlight 'App loaded preserved data on startup ❎'

**Review by muneebulrehman — APPROVED** [review]
Hi @UbdaNam , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- Good follow-up with the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**